.github/workflows/build.yml

```diff
@@ -21,7 +22,8 @@ jobs:
       - name: Install Go
         uses: actions/setup-go@v5
         with:
-          go-version: '1.23'
```

Comment:
Is go version not needed anymore?

Reply:
I switched to `go-version-file: 'go.mod'` to let `setup-go` action read Go version dynamically from `go.mod` file to reduce manual toil on syncing Go versions between multiple files and have one single "source of truth" for Go version
![image](https://github.com/user-attachments/assets/79d7b1cf-6459-4bd6-95cd-6d4066cdd567)